test/qa_settings.cpp

```diff
@@ -524,6 +530,82 @@ const boost::ut::suite AnnotationTests = [] {
+        std::ignore = s.set({ { "name", "TestNameNew" }, { "scaling_factor", 43.f } }, ctx1);
+
+        expect(eq(std::get<float>(*s.get("scaling_factor")), 43.f));       // get the latest value
+        expect(eq(std::get<float>(*s.get("scaling_factor", ctx1)), 43.f)); // get same value, but over the context
+        expect(eq(std::get<float>(*s.get("scaling_factor", ctx0)), 42.f)); // get value with an older timestamp
+    };
+
+    auto matchPred = [](const auto &lhs, const auto &rhs) {
+        return !lhs.empty() && std::all_of(lhs.begin(), lhs.end(), [&](const auto &v) { return rhs.contains(v.first) && rhs.at(v.first) == v.second; });
+    };
+
+    "CtxSettings Parsing"_test = [&] {
+        graph      flow_graph;
+        auto      &block = flow_graph.make_node<TestBlock<float>>({ { "name", "TestName" }, { "scaling_factor", 2.f } });
+        auto       s     = ctx_settings(block, matchPred);
+        const auto ctx0  = SettingsCtx(std::chrono::system_clock::now(), { { "BPCID", 1 } });
+        std::ignore      = s.set({ { "name", "TestNameAlt" }, { "scaling_factor", 42.f } }, ctx0);
+        const auto ctx1  = SettingsCtx(std::chrono::system_clock::now(), { { "BPCID", 1 }, { "SID", 2 } });
+        std::ignore      = s.set({ { "name", "TestNameNew" }, { "scaling_factor", 43.f } }, ctx1);
+
+        // exact matches for contexts work
+        expect(eq(std::get<float>(*s.get("scaling_factor", ctx0)), 42.f));
+        expect(eq(std::get<float>(*s.get("scaling_factor", ctx1)), 43.f));
+
+        // matching by using the custom predicate
+        auto ctx3 = SettingsCtx(std::chrono::system_clock::now(), { { "BPCID", 1 }, { "SID", 2 }, { "BPID", 1 } });
+        expect(eq(std::get<float>(*s.get("scaling_factor", ctx3)), 43.f));
+        auto ctx4   = SettingsCtx(std::chrono::system_clock::now(), { { "BPCID", 1 }, { "SID", 2 }, { "BPID", 3 } });
+        std::ignore = s.set({ { "scaling_factor", 44.f } }, ctx4);
+        expect(eq(std::get<float>(*s.get("scaling_factor", ctx4)), 44.f));
+
+        // doesn't exist
+        auto ctx6 = SettingsCtx(std::chrono::system_clock::now(), { { "BPCID", 9 }, { "SID", 9 }, { "BPID", 9 }, { "GID", 9 } });
+        expect(s.get("scaling_factor", ctx6) == std::nullopt);
+    };
+
+    "CtxSettings Drop-In Settings replacement"_test = [&] {
+        // the multiplexed Settings can be used as a drop-in replacement for "normal" Settings
+        graph flow_graph;
+        auto &block = flow_graph.make_node<TestBlock<float>>({ { "name", "TestName" }, { "scaling_factor", 2.f } });
+        auto  s     = std::make_unique<ctx_settings<std::remove_reference<decltype(block)>::type>>(block, matchPred);
+        block.set_settings(s);
+        auto ctx0   = SettingsCtx(std::chrono::system_clock::now());
+        std::ignore = block.settings().set({ { "name", "TestNameAlt" }, { "scaling_factor", 42.f } }, ctx0);
+        expect(eq(std::get<float>(*block.settings().get("scaling_factor")), 42.f));
+    };
```

Comment:
As discussed above, could you also add the FAIR specific use-case and matcher that would implement the following scenario implemented [here](https://github.com/fair-acc/opencmw-cpp/pull/156), notably:

```cpp
opencmw::CtxSetting<int, std::string, 16, std::chrono::seconds, 3600 * 24, 10> settings; // history length of 24 hours, time-out of 10 seconds

assert(settings.commit(TimingCtx(1, 1, 1, 1), 101)); // store 101 for 'TimingCtx(cid=1, sid=1, pid=1, gid=1)'
assert(settings.commit(TimingCtx(1, 1, 1), 102));    // store 102 for 'TimingCtx(cid=1, sid=1, pid=1, gid=-1)'
assert(settings.commit(TimingCtx(1, 1), 103));       // store 103 for 'TimingCtx(cid=1, sid=1, pid=-1, gid=-1)'
assert(settings.commit(TimingCtx(1), 104));          // store 104 for 'TimingCtx(cid=1, sid=-1, pid=-1, gid=-1)'

// retrieve settings
assert(settings.get(TimingCtx(1)) == 104);
assert(settings.get(TimingCtx(1, 1, 1, 1)) == 101);
assert(settings.get(TimingCtx(1, 1, 1, 1)).timingCtx == TimingCtx(1, 1, 1, 1));
assert(settings.get(TimingCtx(1, 1, 1, 1)) == TimingCtx(1, 1, 1, 1)); // short-hand notation
assert(settings.get(TimingCtx(1, 1, 1, 2)) == 102); // no setting for 'gid=2' -> fall back to 'gid=-1'
assert(settings.get(TimingCtx(1, 1, 1, 2)) == TimingCtx(1, 1, 1));
assert(settings.get(TimingCtx(1, 1, 2, 2)) == 103); // no setting for 'pid=2' and 'gid=2' -> fall back to 'pid=gid=-1'
assert(settings.get(TimingCtx(1, 1, 2, 2)) == TimingCtx(1, 1));

// stage new settings
auto [committed, commitTimeStamp] = settings.commit(TimingCtx(1), 55); // immediately store 55 for 'TimingCtx(cid=1)'
assert(not settings.stage(TimingCtx(1), 555, "tokenNo1"));
assert(not settings.stage(TimingCtx(1), 556, "tokenNo2"));

assert(settings.get(TimingCtx(1)) == 55); // staged settings not yet commited
assert(settings.get(TimingCtx(2)) != TimingCtx(2));
assert(settings.commit("tokenNo1"));
assert(settings.get(TimingCtx(1)) == 555);
assert(settings.commit("tokenNo2"));
assert(settings.get(TimingCtx(1)) == 556);
```

Many thanks in advance.